dali/pipeline/util/thread_pool.cc

```diff
@@ -79,14 +83,25 @@ void ThreadPool::WaitForWork(bool checkForErrors) {
     for (size_t i = 0; i < threads_.size(); ++i) {
       if (!tl_errors_[i].empty()) {
         // Throw the first error that occurred
-        string error = "Error in thread " + std::to_string(i) + ": " + tl_errors_[i].front();
+        string error = make_string("Error in thread ", i, ": ", tl_errors_[i].front());
         tl_errors_[i].pop();
         throw std::runtime_error(error);
       }
     }
   }
 }
 
+void ThreadPool::RunAll(bool wait) {
+  {
+    std::lock_guard<std::mutex> lock(mutex_);
+    adding_work_ = false;
+  }
+  condition_.notify_all();
```

Comment:
I think it can be `notify_one()`. Worker threads already have a mechanism to notify a next worker if there is a task pending. In case the number of jobs is smaller than number of workers `notify_all()` would wake more threads than needed.